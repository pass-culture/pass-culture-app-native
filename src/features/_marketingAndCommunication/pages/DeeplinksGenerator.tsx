import { t } from '@lingui/macro'
import AsyncStorage from '@react-native-async-storage/async-storage'
import React, { useCallback, useState } from 'react'
import styled from 'styled-components/native'

import {
  DeeplinksGeneratorForm,
  GeneratedDeeplink,
} from 'features/_marketingAndCommunication/components/DeeplinksGeneratorForm'
import { DeeplinksHistory } from 'features/_marketingAndCommunication/components/DeeplinksHistory'
import { DeeplinksResult } from 'features/_marketingAndCommunication/components/DeeplinksResult'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { getSpacing, Spacer } from 'ui/theme'

const linksInitialState: Array<GeneratedDeeplink> = []

export const DeeplinksGenerator = () => {
  const [result, setResult] = useState(linksInitialState[0])
  const [links, setLinks] = useState<Array<GeneratedDeeplink>>(linksInitialState)
  const [keepHistory, setKeepHistory] = useState(false)
  const { showErrorSnackBar } = useSnackBarContext()

  const onGenerate = useCallback(
    (generatedDeeplink) => {
      async function add() {
        setLinks((previousGeneratedLinks) => [generatedDeeplink, ...previousGeneratedLinks])
        setResult(generatedDeeplink)
        if (keepHistory) {
          try {
            const localHistory = JSON.parse((await AsyncStorage.getItem('mac_history')) || '[]')
            await AsyncStorage.setItem(
              'mac_history',
              JSON.stringify([...localHistory, generatedDeeplink])
            )
          } catch (error) {
            if (error instanceof Error)
              showErrorSnackBar({
                message: t`Le lien ${generatedDeeplink.universalLink} n'a pas été ajouté à l'historique: ${error.message}`,
                timeout: SNACK_BAR_TIME_OUT,
              })
          }
        }
      }
      add()
    },
    [keepHistory]
  )

  return (
    <React.Fragment>
      <PageHeader title={t`Envie de tout envie de lien\u00a0?`} />
      <Spacer.TopScreen />
      <Spacer.Column numberOfSpaces={18} />
      <Row>
        <Left>
          <DeeplinksGeneratorForm onCreate={onGenerate} />
        </Left>
        <Right>
          <DeeplinksResult result={result} />
          <Divider />
          <DeeplinksHistory
            history={links}
            keepHistory={keepHistory}
            setKeepHistory={setKeepHistory}
            rehydrateHistory={(history) => setLinks(history)}
          />
        </Right>
      </Row>
      <Spacer.Column numberOfSpaces={4} />
    </React.Fragment>
  )
}

const Divider = styled.View(({ theme }) => ({
  height: getSpacing(2),
  backgroundColor: theme.colors.greyLight,
}))

const Row = styled.View(({ theme }) => ({
  flex: 1,
  flexDirection: theme.isMobileViewport ? 'column' : 'row',
}))

const Left = styled.View({
  flex: 1,
})

const Right = styled.View({
  flex: 1,
})
