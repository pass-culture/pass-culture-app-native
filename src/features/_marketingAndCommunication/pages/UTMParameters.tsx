import React from 'react'
import { Text } from 'react-native'
import styled from 'styled-components/native'

import { useUtmParams } from 'libs/utm'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { getSpacing, Spacer } from 'ui/theme'

export const UTMParameters = () => {
  const { campaign, source, medium, campaignDate } = useUtmParams()

  return (
    <React.Fragment>
      <PageHeader title="UTM parameters" />
      <Spacer.TopScreen />
      <Spacer.Column numberOfSpaces={15} />
      <Container>
        <AlignedText>
          <Text>traffic_campaign: {campaign}</Text>
        </AlignedText>
        <Spacer.Column numberOfSpaces={2} />
        <AlignedText>
          <Text>traffic_medium: {medium}</Text>
        </AlignedText>
        <Spacer.Column numberOfSpaces={2} />
        <AlignedText>
          <Text>traffic_source: {source}</Text>
        </AlignedText>
        <Spacer.Column numberOfSpaces={2} />
        <AlignedText>
          <Text>campaign_date: {campaignDate?.toLocaleString()}</Text>
        </AlignedText>
      </Container>
    </React.Fragment>
  )
}

const Container = styled.View({
  paddingHorizontal: getSpacing(5),
})

const AlignedText = styled.View({
  flexDirection: 'row',
  alignItems: 'center',
})
