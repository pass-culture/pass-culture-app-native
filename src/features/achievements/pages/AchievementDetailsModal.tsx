import React, { useEffect } from 'react'
import styled, { useTheme } from 'styled-components/native'

import { AchievementEnum } from 'api/gen'
import { useAchievementDetails } from 'features/achievements/hooks/useAchievementDetails'
import { AppInformationModal } from 'ui/components/modals/AppInformationModal'
import { getSpacing, Typo, Spacer } from 'ui/theme'

interface Props {
  visible: boolean
  hideModal: () => void
  name: AchievementEnum
}

export const AchievementDetailsModal = ({ visible, hideModal, name }: Props) => {
  const achievement = useAchievementDetails(name)
  const theme = useTheme()

  useEffect(() => {
    if (!visible) return
    achievement?.track()
  }, [achievement, visible])

  if (!achievement) return null

  const Illustration = achievement.illustration

  return (
    <AppInformationModal
      title=""
      visible={visible}
      onCloseIconPress={hideModal}
      testIdSuffix="achievement-details">
      <Container>
        <IconsWrapper>
          <Illustration size={theme.illustrations.sizes.fullPage} />
        </IconsWrapper>
        <Spacer.Column numberOfSpaces={6} />
        <BodyWrapper isCompleted={achievement.completed}>
          {achievement.completed ? (
            <StyledBody>Fait le {achievement.completedAt}</StyledBody>
          ) : (
            <Typo.Body>Non débloqué</Typo.Body>
          )}
        </BodyWrapper>
        <Spacer.Column numberOfSpaces={4} />
        {achievement.completed ? (
          <React.Fragment>
            <Typo.Title3>{achievement.title}</Typo.Title3>
            <Spacer.Column numberOfSpaces={4} />
          </React.Fragment>
        ) : null}
        <StyledDescrption>{achievement.description}</StyledDescrption>
      </Container>
    </AppInformationModal>
  )
}

const Container = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
  padding: getSpacing(6),
})
const IconsWrapper = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
})

const BodyWrapper = styled.View<{ isCompleted: boolean }>(({ isCompleted, theme }) => ({
  backgroundColor: isCompleted ? theme.colors.primaryDark : theme.colors.greyLight,
  paddingHorizontal: getSpacing(2),
  paddingVertical: getSpacing(1),
  borderRadius: getSpacing(1),
}))

const StyledBody = styled(Typo.Body)(({ theme }) => ({
  color: theme.colors.white,
}))

const StyledDescrption = styled(Typo.Body)({
  textAlign: 'center',
})
