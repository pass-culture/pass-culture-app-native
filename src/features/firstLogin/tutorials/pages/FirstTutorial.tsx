import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'

import { GenericTutorial } from 'features/firstLogin/tutorials/components/GenericTutorial'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import TutorialPassLogo from 'ui/animations/tutorial_pass_logo.json'

import { TutorialSwiper } from '../components/TutorialSwiper'

export function FirstTutorial() {
  const { navigate } = useNavigation<UseNavigationType>()

  function onSwipeLeft() {
    navigate('SecondTutorial')
  }

  function onButtonPress() {
    navigate('SecondTutorial')
  }

  return (
    <TutorialSwiper onSwipeLeft={onSwipeLeft}>
      <GenericTutorial
        animation={TutorialPassLogo}
        buttonCallback={onButtonPress}
        buttonText={_(t`Continuer`)}
        pauseAnimationOnRenderAtFrame={62}
        step={1}
        subTitle={_(t`c'est...`)}
        text={_(t`une initiative financée par le Ministère de la Culture.`)}
        title={_(t`Le pass Culture`)}
      />
    </TutorialSwiper>
  )
}
