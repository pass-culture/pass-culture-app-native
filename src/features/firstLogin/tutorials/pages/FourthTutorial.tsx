import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'

import { GenericTutorial } from 'features/firstLogin/tutorials/components/GenericTutorial'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import StarAnimation from 'ui/animations/tutorial_star.json'

export function FourthTutorial() {
  const { navigate } = useNavigation<UseNavigationType>()

  function onSwipeRight() {
    navigate('ThirdTutorial')
  }

  function onButtonPress() {
    navigate('TabNavigator')
  }

  return (
    <GenericTutorial
      animation={StarAnimation}
      buttonCallback={onButtonPress}
      buttonText={_(t`Découvrir`)}
      pauseAnimationOnRenderAtFrame={62}
      onSwipeRight={onSwipeRight}
      step={4}
      subTitle={_(t`quotidiennes`)}
      text={_(
        t`Nos nombreux partenaires ajoutent de nouvelles offres quotidiennement. Découvre les dès maintenant !`
      )}
      title={_(t`Des nouveautés`)}
    />
  )
}
