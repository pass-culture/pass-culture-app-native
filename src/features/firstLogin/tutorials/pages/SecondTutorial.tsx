import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'

import { GenericTutorial } from 'features/firstLogin/tutorials/components/GenericTutorial'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import TutorialOffers from 'ui/animations/tutorial_offers.json'

export function SecondTutorial() {
  const { navigate } = useNavigation<UseNavigationType>()

  function onSwipeLeft() {
    navigate('ThirdTutorial')
  }

  function onSwipeRight() {
    navigate('FirstTutorial')
  }

  function onButtonPress() {
    navigate('ThirdTutorial')
  }

  return (
    <GenericTutorial
      animation={TutorialOffers}
      buttonCallback={onButtonPress}
      onSwipeLeft={onSwipeLeft}
      onSwipeRight={onSwipeRight}
      buttonText={_(t`Continuer`)}
      pauseAnimationOnRenderAtFrame={62}
      step={2}
      subTitle={_(t`et si tu es...`)}
      text={_(
        t`dans l’année de tes 18 ans, obtiens l’aide financière pass Culture d’un montant de 300€ à dépenser dans l’application.`
      )}
      title={_(t`Des offres pour tous`)}
    />
  )
}
