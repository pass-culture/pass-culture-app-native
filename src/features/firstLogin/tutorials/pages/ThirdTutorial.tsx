import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'

import { GenericTutorial } from 'features/firstLogin/tutorials/components/GenericTutorial'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { useGeolocation } from 'libs/geolocation'
import { _ } from 'libs/i18n'
import GeolocationAnimation from 'ui/animations/geolocalisation.json'

export function ThirdTutorial() {
  const { navigate } = useNavigation<UseNavigationType>()
  const { requestGeolocPermission } = useGeolocation()

  function onSwipeLeft() {
    navigate('FourthTutorial')
  }

  function onSwipeRight() {
    navigate('SecondTutorial')
  }

  async function onGeolocationButtonPress() {
    await requestGeolocPermission({
      onSubmit: () => {
        navigate('FourthTutorial')
      },
    })
  }

  return (
    <GenericTutorial
      animation={GeolocationAnimation}
      buttonCallback={onGeolocationButtonPress}
      buttonText={_(t`Activer la géolocalisation`)}
      onSwipeLeft={onSwipeLeft}
      onSwipeRight={onSwipeRight}
      pauseAnimationOnRenderAtFrame={62}
      step={3}
      subTitle={_(t`à portée de main !`)}
      text={_(
        t`Active la géolocalisation pour découvrir toutes les offres existantes autour de toi.`
      )}
      title={_(t`Toute la culture`)}
    />
  )
}
