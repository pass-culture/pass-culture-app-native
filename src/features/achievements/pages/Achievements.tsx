import { useRoute } from '@react-navigation/native'
import React, { FC, useEffect } from 'react'
import { View } from 'react-native'
import { FlatList } from 'react-native-gesture-handler'
import { useTheme } from 'styled-components'
import styled from 'styled-components/native'

import { AchievementEnum } from 'api/gen'
import { Achievement } from 'features/achievements/components/Achievement'
import {
  achievementCategoryDisplayTitles,
  achievementData,
} from 'features/achievements/data/AchievementData'
import { getAchievements } from 'features/achievements/helpers/getAchievements'
import { useAuthContext } from 'features/auth/context/AuthContext'
import { UseRouteType } from 'features/navigation/RootNavigator/types'
import { ProgressBar } from 'ui/components/bars/ProgressBar'
import { ViewGap } from 'ui/components/ViewGap/ViewGap'
import { SecondaryPageWithBlurHeader } from 'ui/pages/SecondaryPageWithBlurHeader'
import { AccessibleIcon } from 'ui/svg/icons/types'
import { getSpacing, Typo } from 'ui/theme'
import { getHeadingAttrs } from 'ui/theme/typographyAttrs/getHeadingAttrs'

const emptyAchievement = {
  name: 'empty' as AchievementEnum,
  isCompleted: false,
  illustration: null as unknown as FC<AccessibleIcon>,
  title: '',
  description: '',
}

export const Achievements = () => {
  const {
    params: { from },
  } = useRoute<UseRouteType<'Achievements'>>()
  const { user } = useAuthContext()
  const { uniqueColors } = useTheme()
  const { categories, track } = getAchievements({
    achievements: achievementData,
    completedAchievements: user?.achievements || [],
  })

  useEffect(() => {
    track(from)
  }, [from, track])

  return (
    <SecondaryPageWithBlurHeader title="Mes succÃ¨s">
      <ViewGap gap={4}>
        {categories.map((category) => {
          const isOddAchievements = category.achievements.length % 2 !== 0
          const achievements = isOddAchievements
            ? [...category.achievements, emptyAchievement]
            : category.achievements

          return (
            <ViewGap gap={4} key={category.name}>
              <AchievementsGroupeHeader>
                <View>
                  <Typo.Title4 {...getHeadingAttrs(2)}>
                    {achievementCategoryDisplayTitles[category.name]}
                  </Typo.Title4>
                  <StyledBody>{category.remainingAchievementsText}</StyledBody>
                </View>
                <CompletionContainer>
                  <ProgressBarContainer>
                    <ProgressBar
                      progress={category.progress}
                      colors={[uniqueColors.brand]}
                      height={2.5}
                    />
                  </ProgressBarContainer>
                  <Typo.BodyS>{category.progressText}</Typo.BodyS>
                </CompletionContainer>
              </AchievementsGroupeHeader>
              <FlatList
                data={achievements}
                numColumns={2}
                keyExtractor={(item) => item.name}
                contentContainerStyle={{
                  gap: getSpacing(6),
                  rowGap: getSpacing(6),
                  paddingHorizontal: getSpacing(2),
                }}
                columnWrapperStyle={{
                  gap: getSpacing(6),
                }}
                renderItem={({ item }) =>
                  item.illustration ? (
                    <Achievement
                      name={item.name}
                      title={item.title}
                      Illustration={item.illustration}
                      isCompleted={item.isCompleted}
                    />
                  ) : (
                    <EmptyAchievement accessible={false} />
                  )
                }
              />
            </ViewGap>
          )
        })}
      </ViewGap>
    </SecondaryPageWithBlurHeader>
  )
}

const EmptyAchievement = styled.View({
  flex: 0.5,
})

const StyledBody = styled(Typo.BodyAccentXs)(({ theme }) => ({
  color: theme.designSystem.color.text.subtle,
}))

const AchievementsGroupeHeader = styled.View({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
})

const CompletionContainer = styled(ViewGap).attrs({ gap: 2 })({
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
  justifyContent: 'flex-end',
})

const ProgressBarContainer = styled.View({
  flex: 1,
  maxWidth: getSpacing(20),
})
