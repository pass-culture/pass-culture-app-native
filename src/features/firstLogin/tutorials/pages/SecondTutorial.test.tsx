import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import { SecondTutorial } from 'features/firstLogin/tutorials/pages/SecondTutorial'

import { navigate } from '../../../../../__mocks__/@react-navigation/native'

describe('SecondTutorial page', () => {
  it('should display animation', async () => {
    const firstTutorial = render(<SecondTutorial />)
    expect(firstTutorial).toMatchSnapshot()
  })

  it('should go to home when clicking skip tutorial button', () => {
    const { getByText } = render(<SecondTutorial />)

    const skipTutorials = getByText('Tout passer')
    fireEvent.press(skipTutorials)

    expect(navigate).toBeCalledWith('TabNavigator')
  })

  it('should navigate to third tutorial when clicking next button', () => {
    const { getByText } = render(<SecondTutorial />)

    const nextButton = getByText('Continuer')
    fireEvent.press(nextButton)

    expect(navigate).toBeCalledWith('ThirdTutorial')
  })
})
