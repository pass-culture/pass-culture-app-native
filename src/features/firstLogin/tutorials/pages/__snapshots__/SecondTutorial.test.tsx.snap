// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondTutorial page should display animation 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      height="100%"
      width="100%"
    >
      undefined-SVG-Mock
    </View>
  </View>,
  <View
    customHeight={16}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flexGrow": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexGrow": 1,
            "maxHeight": 900,
            "maxWidth": 500,
            "paddingHorizontal": 20,
            "paddingTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 24,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "maxWidth": 500,
              "opacity": 1,
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
              "width": "100%",
            }
          }
          testID="button-container"
        >
          <Text
            adjustsFontSizeToFit={false}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#626262",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 15,
                  "lineHeight": 20,
                  "marginLeft": 5,
                  "maxWidth": 650,
                },
              ]
            }
            testID="button-title"
            textColor="#626262"
          >
            Tout passer
          </Text>
        </View>
      </View>
      <View
        config={
          Object {
            "directionalOffsetThreshold": 400,
            "gestureIsClickThreshold": 0.1,
            "velocityThreshold": 0.03,
          }
        }
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onSwipeLeft={[Function]}
        onSwipeRight={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexGrow": 1,
              "width": "100%",
            },
          ]
        }
      >
        <View
          flex={2}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
              },
            ]
          }
        />
        <View>
          undefined-Lottie-Mock
        </View>
        <View
          flex={1}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-ExtraBoldItalic",
                "fontSize": 28,
                "lineHeight": 34,
                "textAlign": "center",
              },
            ]
          }
        >
          Des offres pour tous
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-MediumItalic",
                "fontSize": 24,
                "lineHeight": 28,
                "textAlign": "center",
              },
            ]
          }
        >
          et si tu es...
        </Text>
        <View
          flex={1}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#151515",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 15,
                "lineHeight": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          dans l’année de tes 18 ans, obtiens l’aide financière pass Culture d’un montant de 300€ à dépenser dans l’application.
        </Text>
        <View
          flex={2}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#eb0055",
            "borderRadius": 24,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "maxWidth": 500,
            "opacity": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
            "width": "100%",
          }
        }
        testID="button-container"
      >
        <Text
          adjustsFontSizeToFit={false}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 5,
                "maxWidth": 650,
              },
            ]
          }
          testID="button-title"
          textColor="#ffffff"
        >
          Continuer
        </Text>
      </View>
      <View
        config={
          Object {
            "directionalOffsetThreshold": 400,
            "gestureIsClickThreshold": 0.1,
            "velocityThreshold": 0.03,
          }
        }
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onSwipeLeft={[Function]}
        onSwipeRight={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": "100%",
            },
          ]
        }
      >
        <View
          numberOfSpaces={8}
          style={
            Array [
              Object {
                "height": 32,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            <View
              fill="#15884f"
              height={8}
              testID="dot-icon"
              width={8}
            >
              dot-icon-SVG-Mock
            </View>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            <View
              fill="#eb0055"
              height={12}
              testID="dot-icon"
              width={12}
            >
              dot-icon-SVG-Mock
            </View>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            <View
              fill="#c7c7cc"
              height={8}
              testID="dot-icon"
              width={8}
            >
              dot-icon-SVG-Mock
            </View>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "width": 8,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            <View
              fill="#c7c7cc"
              height={8}
              testID="dot-icon"
              width={8}
            >
              dot-icon-SVG-Mock
            </View>
          </View>
        </View>
        <View
          numberOfSpaces={8}
          style={
            Array [
              Object {
                "height": 32,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      customHeight={8}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
  </View>,
]
`;
